pandas/core/strings.py

```diff
@@ -1095,24 +1095,83 @@ def str_pad(arr, width, side='left', fillchar=' '):
 
 def str_split(arr, pat=None, n=None):
     """
-    Split each string (a la re.split) in the Series/Index by given
-    pattern, propagating NA values. Equivalent to :meth:`str.split`.
+    Split strings around given separator/delimiter.
+
+    Split each str in the caller's values by given
+    pattern, propagating NaN values. Equivalent to :meth:`str.split`.
 
     Parameters
     ----------
     pat : string, default None
-        String or regular expression to split on. If None, splits on whitespace
+        String or regular expression to split on.
+        If `None`, split on whitespace.
     n : int, default -1 (all)
-        None, 0 and -1 will be interpreted as return all splits
+        Vary dimensionality of output.
+
+        * `None`, 0 and -1 will be interpreted as return all splits
```

Comment:
I would not put this in a bullet point, as this breaks the flow (i.e. need to have a blank line for the list to render..). Lists with one bullet point are a bit strange anyhow